src/osd/ReplicatedBackend.cc

```diff
@@ -832,11 +833,11 @@ void ReplicatedBackend::be_deep_scrub(
     dout(25) << __func__ << "  " << poid << " got "
 	     << r << " on omap scan, read_error" << dendl;
     o.read_error = true;
-  }
-
+  } else {
```

Comment:
An omap_get_header() EIO could still get here if omap_get_iterator() doesn't error.  A return whenever setting read_error would handle the 3 error cases here.